models/ProfileModel.js

```diff
@@ -122,27 +122,58 @@ const getUserLikes = `
   )
   ORDER BY p.timestamp DESC`;
 
-const getProfileContents = `
-  SELECT 
-    (SELECT COUNT(*) FROM
-      post WHERE "userId" = a."userId"
-    ) AS "postCount",
-    a."bio",
-    a."joinedDate",
-    a."displayName",
-    a."username",
-    (SELECT COUNT(*) FROM 
-      follow WHERE "followedUserId" = a."userId"
-    ) AS "followerCount",
-    (SELECT COUNT(*) FROM 
-      follow WHERE "followerUserId" = a."userId"
-    ) AS "followingCount"
-  FROM app_user AS a
-  WHERE a."username" = $1`;
+const getProfileContents = `SELECT 
+(SELECT COUNT(*) FROM post WHERE "userId" = a."userId") AS "postCount",
+a."bio",
+a."joinedDate",
+a."displayName",
+a."username",
+(SELECT COUNT(*) FROM follow WHERE "followedUserId" = a."userId" AND "followStatus" = TRUE ) AS "followerCount",
+(SELECT COUNT(*) FROM follow WHERE "followerUserId" = a."userId" AND "followStatus" = TRUE ) AS "followingCount",
+f."followStatus" AS "followStatus"
+FROM app_user AS a
+LEFT JOIN follow as f ON f."followerUserId" = $1 AND f."followedUserId" = a."userId"
+WHERE a."username" = $2`;
+
+const getFollowStatus = `SELECT 
+CASE 
+    WHEN EXISTS (
+        SELECT 1
+        FROM follow
+        WHERE "followerUserId" = $1
+          AND "followedUserId" = (SELECT "userId" FROM app_user WHERE "username" = $2)
+          AND "followStatus" = TRUE
+    ) THEN TRUE
+    ELSE FALSE
+END AS "followStatus";`;
```

Comment:
Instead of wrapping in a CASE statement I think we can just grab the `followStatus` like below. I tested it out and technically this can return nothing if the username isn't in the database, but that shouldn't ever happen. Best practice would be to throw an error though, so in the controller maybe check if `followStatus` is undefined and if so throw an error. Dunno how detailed we want to be with our error handling but if we're lazy we can make another ticket an throw it into the void lol

```suggestion
const getFollowStatus = `SELECT 
"followStatus"
FROM follow
WHERE "followerUserId" = $1
AND "followedUserId" = (SELECT "userId" FROM app_user WHERE "username" = $2);`;
```

Reply:
Not going to lie, this model is not even being used anymore, its outdated code because we grab the followStatus within `getProfileContents`